dbms/src/TestUtils/FunctionTestUtils.cpp

```diff
@@ -115,39 +115,48 @@ void blockEqual(
     }
 }
 
+std::pair<ExpressionActionsPtr, String> buildFunction(Context & context, const String & func_name, const ColumnsWithTypeAndName & columns, const ColumnNumbers & argument_column_numbers, const TiDB::TiDBCollatorPtr & collator)
+{
+    std::vector<NameAndTypePair> source_columns;
+    for (const auto & column : columns)
+        source_columns.emplace_back(column.name, column.type);
+    DAGExpressionAnalyzer analyzer(source_columns, context);
+    ExpressionActionsChain chain;
+    auto last_step = analyzer.initAndGetLastStep(chain);
+    auto result_name = DB::FunctionBuildHelper::buildFunction(ColumnsFunctionExpr(columns, argument_column_numbers, func_name, collator), &analyzer, last_step.actions);
+    return std::make_pair(last_step.actions, result_name);
```

Comment:
`last_step.setRequireOutput(result_name);`
and then
`chain.finalize()`

Reply:
Good idea, done.